lib/lib.go

```diff
@@ -0,0 +1,229 @@
+		return nil, err
+	}
+
+	fmttr, err := formatters.NewForConfig(cfg.ReadOnly())
+	if err != nil {
+		return nil, err
+	}
+
+	rs := ResolveSubmitter(pyxisClient, cfg.ReadOnly())
+
+	return &CheckContainerRunner{
+		Cfg:       cfg,
+		Pc:        pyxisClient,
+		Eng:       engine,
+		Formatter: fmttr,
+		Rw:        &runtime.ResultWriterFile{},
+		Rs:        rs,
+	}, nil
+}
+
+// checkOperatorRunner contains all of the components necessary to run checkOperator.
+type CheckOperatorRunner struct {
+	Cfg       *runtime.Config
+	Eng       engine.CheckEngine
+	Formatter formatters.ResponseFormatter
+	Rw        ResultWriter
+}
+
+// newCheckOperatorRunner returns a checkOperatorRunner containing all of the tooling necessary
+// to run checkOperator.
+func NewCheckOperatorRunner(ctx context.Context, cfg *runtime.Config) (*CheckOperatorRunner, error) {
+	cfg.Policy = policy.PolicyOperator
+	cfg.Submit = false // there's no such thing as submitting for operators today.
+
+	engine, err := engine.NewForConfig(ctx, cfg.ReadOnly())
+	if err != nil {
+		return nil, err
+	}
+
+	fmttr, err := formatters.NewForConfig(cfg.ReadOnly())
+	if err != nil {
+		return nil, err
+	}
+
+	return &CheckOperatorRunner{
+		Cfg:       cfg,
+		Eng:       engine,
+		Formatter: fmttr,
+		Rw:        &runtime.ResultWriterFile{},
+	}, nil
+}
+
+// resolveSubmitter will build out a resultSubmitter if the provided pyxisClient, pc, is not nil.
+// The pyxisClient is a required component of the submitter. If pc is nil, then a noop submitter
+// is returned instead, which does nothing.
+func ResolveSubmitter(pc PyxisClient, cfg certification.Config) ResultSubmitter {
+	if pc != nil {
+		return &ContainerCertificationSubmitter{
+			CertificationProjectID: cfg.CertificationProjectID(),
+			Pyxis:                  pc,
+			DockerConfig:           cfg.DockerConfig(),
+			PreflightLogFile:       cfg.LogFile(),
+		}
+	}
+	return NewNoopSubmitter(true, "", nil)
+}
+
+// GetContainerPolicyExceptions will query Pyxis to determine if
+// a given project has a certification excemptions, such as root or scratch.
+// This will then return the corresponding policy.
+//
+// If no policy exception flags are found on the project, the standard
+// container policy is returned.
+func GetContainerPolicyExceptions(ctx context.Context, pc PyxisClient) (policy.Policy, error) {
+	certProject, err := pc.GetProject(ctx)
+	if err != nil {
+		return "", fmt.Errorf("could not retrieve project: %w", err)
+	}
+	// log.Debugf("Certification project name is: %s", certProject.Name)
+	if certProject.Container.Type == "scratch" {
+		return policy.PolicyScratch, nil
+	}
+
+	// if a partner sets `Host Level Access` in connect to `Privileged`, enable RootExceptionContainerPolicy checks
+	if certProject.Container.Privileged {
+		return policy.PolicyRoot, nil
+	}
+	return policy.PolicyContainer, nil
+}
+
+// PreflightCheck executes checks, interacts with pyxis, format output, writes, and submits results.
+func PreflightCheck(
```

Comment:
Not sure we need this exported.

Reply:
This is how I kick off the tests right now.  But if we go with your proposal this could all be kept internal.

How I kick it off for example:
```
err = plib.PreflightCheck(context.TODO(), runner.Cfg, runner.Pc, runner.Eng, runner.Formatter, runner.Rw, runner.Rs)
		if err != nil {
			return err
		}
```